### PR TITLE
fix: Custom theme for embedded components in express build mode

### PR DESCRIPTION
**Pull request by mshabarov**
Fixes https://github.com/vaadin/flow/issues/15873

### CODE REVIEW

File: flow-server/src/main/java/com/vaadin/flow/server/BootstrapHandler.java [hunk 1]
```diff
@@ -1595,6 +1597,72 @@ protected static void addJavaScriptEntryPoints(
...
+     * @param fileName
+     *            the stylesheet file name to add a reference to
+     * @return the collection of link tag elements to be added to the page
+     * @throws IOException
+     *             if theme name cannot be extracted from file
+     */
+    protected static Collection<Element> getTagForTheme(
+            DeploymentConfiguration config, String fileName)
+            throws IOException {
+        Collection<Element> tags = new ArrayList<>();
```
Comment (caalador) [inline flow-server/src/main/java/com/vaadin/flow/server/BootstrapHandler.java:1616]: I would guess calling this List here would perhaps not confuse sonatype to think we return mutable and immutable collections?

File: flow-server/src/main/java/com/vaadin/flow/server/BootstrapHandler.java [hunk 2]
```diff
@@ -1595,6 +1597,72 @@ protected static void addJavaScriptEntryPoints(
...
+            for (File themeFolder : Objects.requireNonNull(
+                    packagedThemesFolder.listFiles(File::isDirectory),
+                    "Expected at least one theme in the front-end generated themes folder")) {
+                String packagedThemeName = themeFolder.getName();
+                packagedThemeNames.add(packagedThemeName);
+                Element tag = getLinkTagWithStyleRef(packagedThemeName,
+                        fileName);
+                tags.add(tag);
+            }
+        }
```
Comment (caalador) [inline flow-server/src/main/java/com/vaadin/flow/server/BootstrapHandler.java:1646]: The packagedThemeNames could just be a `boolean customThemeFound = false;` with the check in the for loop `customThemeFound = customThemeFound && packagedThemeName.equals(themeName.get));` as that is the only thing the list is used for.

File: flow-server/src/test/java/com/vaadin/flow/server/communication/IndexHtmlRequestHandlerTest.java [hunk 3]
```diff
@@ -56,6 +59,7 @@
 import com.vaadin.flow.internal.UsageStatistics;
 import com.vaadin.flow.server.AppShellRegistry;
 import com.vaadin.flow.server.BootstrapHandler;
+import com.vaadin.flow.server.Constants;
```
Comment (caalador) [inline flow-server/src/test/java/com/vaadin/flow/server/communication/IndexHtmlRequestHandlerTest.java:62]: Extra import as the constants are already as static imports?

File: flow-server/src/main/java/com/vaadin/flow/server/BootstrapHandler.java (hunk 1 above)
Comment (mshabarov) [reply flow-server/src/main/java/com/vaadin/flow/server/BootstrapHandler.java]: Replaced by `List`

File: flow-server/src/main/java/com/vaadin/flow/server/BootstrapHandler.java (hunk 2 above)
Comment (mshabarov) [reply flow-server/src/main/java/com/vaadin/flow/server/BootstrapHandler.java]: Changed

File: flow-server/src/test/java/com/vaadin/flow/server/communication/IndexHtmlRequestHandlerTest.java (hunk 3 above)
Comment (mshabarov) [reply flow-server/src/test/java/com/vaadin/flow/server/communication/IndexHtmlRequestHandlerTest.java:60]: I added Constant class explicitly for better readability.

**Review by caalador — DISMISSED** [review]

**Review by caalador — APPROVED** [review]

**Pull request merged**